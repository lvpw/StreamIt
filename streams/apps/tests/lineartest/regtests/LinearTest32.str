/** 
 * Simple program to test finding duplicate computation terms
 * among 
 **/

void->void pipeline LinearTest32 {
  add SimpleSource();
  add RedundantFilter();
  add RedundantPeekFilter();
  add SimpleSink();
}

/** simple source that spits out .1, 1.1, 2.1, ... 99.1 and then repeats. **/
void->float stateful filter SimpleSource {
  int x = 0;
  work push 1 {
    //float t = x + 0.1;
    //t = x;
    push (x);
    x = (x+1)%4;
  }
}

/** simple, printing sink for floating point data. **/
float->void filter SimpleSink {
  work pop 1 {
    println(pop());
  }
}
  
/** simple filter with some redundant computation. **/
float->float filter RedundantFilter {
  work pop 3 push 3 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    push(3*t3 + 5*t2 + 7*t1);
    push(2*t3 + 5*t2 + 8*t1);
    push(1*t3 + 4*t2 + 7*t1);
  }
}

/** a simple filter with redundant peeking computation. **/
float->float filter RedundantPeekFilter {
  work pop 1 peek 3 push 3 {
    float t1 = peek(0);
    float t2 = peek(1);
    float t3 = peek(2);
    pop();
    push(6*t3 + 6*t2 + 3*t1);
    push(5*t3 + 5*t2 + 2*t1);
    push(4*t3 + 4*t2 + 1*t1);
  }
}

