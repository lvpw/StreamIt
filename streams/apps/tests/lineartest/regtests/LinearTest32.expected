LinearWrapper_1.matrix
[[20.0+0.0i 25.0+0.0i 30.0+0.0i 12.0+0.0i 15.0+0.0i 18.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [40.0+0.0i 50.0+0.0i 60.0+0.0i 20.0+0.0i 25.0+0.0i 30.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [60.0+0.0i 75.0+0.0i 90.0+0.0i 28.0+0.0i 35.0+0.0i 42.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [1.0+0.0i 2.0+0.0i 3.0+0.0i 6.0+0.0i 9.0+0.0i 12.0+0.0i 15.0+0.0i 21.0+0.0i 27.0+0.0i]
 [4.0+0.0i 8.0+0.0i 12.0+0.0i 21.0+0.0i 30.0+0.0i 39.0+0.0i 41.0+0.0i 55.0+0.0i 69.0+0.0i]
 [7.0+0.0i 14.0+0.0i 21.0+0.0i 36.0+0.0i 51.0+0.0i 66.0+0.0i 67.0+0.0i 89.0+0.0i 111.0+0.0i]]
LinearWrapper_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]]
RedundantFilter_1.matrix
[[1.0+0.0i 2.0+0.0i 3.0+0.0i]
 [4.0+0.0i 5.0+0.0i 5.0+0.0i]
 [7.0+0.0i 8.0+0.0i 7.0+0.0i]]
RedundantFilter_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i]]
RedundantPeekFilter_1.matrix
[[4.0+0.0i 5.0+0.0i 6.0+0.0i]
 [4.0+0.0i 5.0+0.0i 6.0+0.0i]
 [1.0+0.0i 2.0+0.0i 3.0+0.0i]]
RedundantPeekFilter_1.vector
[[0.0+0.0i 0.0+0.0i 0.0+0.0i]]
